apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/OpenOrders.tsx

```diff
@@ -349,18 +491,71 @@ const ModalContent = ({
   assetFromLabel,
   assetToIcon,
   assetToLabel,
+  assetPrice,
+  assetAmount,
+  totalReturned,
+  orderUtxo,
+  collateralUtxo,
+  bech32Address,
 }: {
   onConfirm: () => void
   onBack: () => void
-  confirmationModal: boolean
   assetFromIcon: React.ReactNode
   assetFromLabel: string
   assetToLabel: string
   assetToIcon: React.ReactNode
+  assetPrice: string
+  assetAmount: string
+  totalReturned: string
+  orderUtxo: string
+  collateralUtxo: string
+  bech32Address: string
 }) => {
   const strings = useStrings()
+
+  const {order} = useSwap()
+  const wallet = useSelectedWallet()
+
+  const [fee, setFee] = useState<string | null>(null)
+
+  const getFee = useCallback(async () => {
```

Comment:
wrap it within a react-query to handle the async state, and pass the dependencies over, leverage the suspense or isLoading, I'd recommend to extract most of this logic that is not UI, into a `/ListOrder/helpers.tsx` and have the hooks and most of the helpers over there.